shuffle/cli/src/deploy.rs

```diff
@@ -1,50 +1,40 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{shared, shared::Home};
+use crate::shared::{
+    build_move_package, get_home_path, normalized_network, read_config, DevApiClient, Home,
+    MAIN_PKG_PATH,
+};
 use anyhow::{anyhow, Result};
 use diem_crypto::PrivateKey;
 use diem_sdk::{
-    client::BlockingClient,
+    client::AccountAddress,
     transaction_builder::TransactionFactory,
     types::{
         transaction::{Module, TransactionPayload},
         LocalAccount,
     },
 };
-use diem_types::{
-    account_state::AccountState, account_state_blob::AccountStateBlob, chain_id::ChainId,
-    transaction::authenticator::AuthenticationKey,
-};
+use diem_types::{chain_id::ChainId, transaction::authenticator::AuthenticationKey};
 use generate_key::load_key;
-use move_binary_format::{file_format::CompiledModule, normalized};
-use move_package::compilation::compiled_package::CompiledPackage;
-use std::{convert::TryFrom, path::Path};
+use serde_json::Value;
+use std::{
+    path::Path,
+    time::{Duration, Instant},
+};
 
 /// Deploys shuffle's main Move Package to the sender's address.
-pub fn handle(project_path: &Path) -> Result<()> {
-    let home = Home::new(shared::get_home_path().as_path())?;
+pub async fn handle(project_path: &Path) -> Result<()> {
+    let network = normalized_network(read_config(project_path)?.get_network())?;
```

Comment:
This will be outdated with your networks.toml PR.

Reply:
Agreed. Will refactor this code after networks.toml PR lands